### PR TITLE
feat: Add tests for api v2 methods

### PR DESCRIPTION
**Pull request by mzaniolo**
Close: #15 
Close: #16
Close: #18

### CODE REVIEW

File: tests/e2e.rs [hunk 1]
```diff
@@ -4,20 +4,46 @@
...
+}
+
+async fn tear_down(zitadel: &mut Zitadel) {
+	let query = ListUsersRequest::new(vec![
+		SearchQuery::new().with_type_query(TypeQuery::new(Userv2Type::Human))
+	]);
+
+	let Ok(mut stream) = zitadel.list_users(query) else {
+		return;
+	};
```
Comment (sirewix) [inline tests/e2e.rs:39]: Is `list_users` returning `Err` ok? shouldn't we throw an error here?

File: tests/e2e.rs [hunk 2]
```diff
@@ -93,26 +116,152 @@ async fn test_e2e_list_human_users() -> Result<()> {
...
-
 	let mut zitadel = Zitadel::new(url, service_account_file.to_path_buf()).await?;
 
 	let idp_links = vec![];
 
-	let user = users::AddHumanUserRequest::new(
-		users::SetHumanProfile::new("John".to_owned(), "list_idp_links".to_owned()),
-		users::SetHumanEmail::new("list_idp_links@doamin.com".to_owned()),
+	let user = AddHumanUserRequest::new(
+		SetHumanProfile::new("John}".to_owned(), "Doe".to_owned()),
```
Comment (sirewix) [inline tests/e2e.rs:124]: `}` typo?

File: tests/e2e.rs [hunk 3]
```diff
@@ -93,26 +116,152 @@ async fn test_e2e_list_human_users() -> Result<()> {
...
+		SearchQuery::new().with_last_name_query(LastNameQuery::new(user_last_name.to_owned()))
+	])
+	.with_asc(true);
+
+	let mut stream = zitadel.list_users(query.clone())?;
+	let mut user_count = 0;
+	while stream.next().await.is_some() {
+		user_count += 1;
+	}
+	assert_eq!(user_count, 2);
```
Comment (sirewix) [inline tests/e2e.rs:247]:
```rust
assert_eq!(zitadel.list_users(query.clone())?.count.await, 2)
```

File: tests/e2e.rs [hunk 4]
```diff
@@ -93,26 +116,152 @@ async fn test_e2e_list_human_users() -> Result<()> {
...
+	while let Some(user) = stream.next().await {
+		let user_name = user
+			.human()
+			.and_then(|u| u.profile())
+			.and_then(|u| u.given_name())
+			.expect("Couldn't get user's given name");
+		assert_ne!(user_name, deleted_user_name);
+		user_count += 1;
+	}
+	assert_eq!(user_count, 1);
```
Comment (sirewix) [inline tests/e2e.rs:262]:
```rust
	let user_count = zitadel.list_users(query.clone())?.inspect(|user| {
		let user_name = user
			.human()
			.and_then(|u| u.profile())
			.and_then(|u| u.given_name())
			.expect("Couldn't get user's given name");
		assert_ne!(user_name, deleted_user_name);
	}).count.await;
	assert_eq!(user_count, 1);
```

**Review by emgrav — APPROVED** [review]

**Review by sirewix — APPROVED** [review]

File: Cargo.toml [hunk 5]
```diff
@@ -129,3 +129,4 @@ verbose_file_reads = "warn"
```
Comment (tlater-famedly) [inline Cargo.toml:129]: Commit title should be prefixed with `test:`, otherwise this ends up in the release notes as a feature.

File: README.md [hunk 6]
```diff
@@ -11,7 +11,11 @@
 [badge-docs-main-img]: https://img.shields.io/badge/docs-main-blue
 [badge-docs-main-url]: https://famedly.github.io/rust-library-template/project_name/index.html
 
-Short description of the project.
```
Comment (tlater-famedly) [inline README.md:14]: Won't block the PR on this, but changes like this should really be a separate commit.

**Review by tlater-famedly — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 5 above)
Comment (tlater-famedly) [reply Cargo.toml:129]: Specifically for the commit that adds tests, of course.

**Pull request merged**